BuildScript/BuildTasks/TestConfigurationsFactory.cs

```diff
@@ -144,11 +144,46 @@ private string GetPlatform (IEnumerable<string> configurationItems)
 
     private ExecutionRuntime GetExecutionRuntime (IEnumerable<string> configurationItems)
     {
-      var kvp = _supportedExecutionRuntimes.Single (configurationItems, () => CreateMissingConfigurationItemException ("execution runtime"));
+      var enumeratedConfigurationItems = configurationItems.ToArray();
+      if (TryGetDockerImageIfIsEnforcedLocalMachine (enumeratedConfigurationItems, out var dockerImageOfEnforcedLocalMachine))
+      {
+        return new ExecutionRuntime (MetadataValueConstants.EnforcedLocalMachine, MetadataValueConstants.EnforcedLocalMachine, false, dockerImageOfEnforcedLocalMachine);
+      }
+
+      var kvp = _supportedExecutionRuntimes.Single (enumeratedConfigurationItems, () => CreateMissingConfigurationItemException ("execution runtime"));
 
       var shouldUseDocker = ShouldUseDocker (kvp.Value);
+      var dockerImage = shouldUseDocker ? kvp.Value : "";
+
+      return new ExecutionRuntime (kvp.Key, kvp.Value, shouldUseDocker, dockerImage);
```

Comment:
We already have a flag, why not always include the docker image?

Reply:
I'm not sure I understand what you mean, but If I do: There are cases where no docker image is tangible, like with `LocalMachine` or `EnforcedLocalMachine` without a docker image (which still works because this change is additive)